test/expect-tests/dune_pkg/fetch_tests.ml

```diff
@@ -202,8 +202,7 @@ let%expect_test "downloading, tarball with no checksum match" =
     Done downloading
     Finished successfully, no checksum verification
     ------
-    files in target dir:
-    plaintext.md |}]
+    files in target dir: |}]
```

Comment:
Is this not missing the file that it should unpack?

Reply:
Does opam support them? All the tarballs I've checked do have a root directory. If opam does indeed support them, we can add support for them as well.

Reply:
I haven't seen them either, because I assume `dune-release` and `opam-publish` create them in a sensible way, but I was curious and yes `opam` supports tarballs without a root: 

If I create a tarball that contains `install.ml` with

```ocaml
let () =
  print_endline "Installation failed";
  exit 10
```

And a repo with `rootless.1`  opam file
```
opam-version: "2.0"
build: [
  ["ocaml" "install.ml"]
]
url {
  src: "file://<path-to>/rootless.tar.gz"
}
```
Then upon `opam install rootless` I get

```
#=== ERROR while compiling rootless.1 =========================================#
# context     2.1.5 | linux/x86_64 | ocaml-base-compiler.4.14.1 | file://<path-to>/rootless-repo
# path        ~/<path-to-switch>/.opam-switch/build/rootless.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build ocaml install.ml
# exit-code   10
# env-file    ~/.opam/log/rootless-602822-ae8c0f.env
# output-file ~/.opam/log/rootless-602822-ae8c0f.out
### output ###
# Installation failed
```

So opam successfully unpacked and failed with the right error message and code as my script.